backend/src/apiserver/resource/resource_manager.go

```diff
@@ -2128,9 +2130,9 @@ func (r *ResourceManager) GetSecret(ctx context.Context, namespace, name string)
 	return secret, nil
 }
 
-// GetSignedUrl retrieves a signed url for the associated artifact.
-func (r *ResourceManager) GetSignedUrl(bucketConfig *objectstore.Config, secret *corev1.Secret, expirySeconds time.Duration, artifactURI string) (string, error) {
-	signedUrl, err := r.objectStore.GetSignedUrl(bucketConfig, secret, expirySeconds, artifactURI)
+// GetSignedUrlWithQueryParams retrieves a signed url for the associated artifact.
+func (r *ResourceManager) GetSignedUrlWithQueryParams(bucketConfig *objectstore.Config, secret *corev1.Secret, expirySeconds time.Duration, artifactURI string, queryParams url.Values) (string, error) {
```

Comment:
Sorry, my previous comment was about keeping `GetSignedURL` and having it call `GetSignedUrlWithQueryParams` to reduce code duplication, so that we don't make an not backwards compatible API change since this is a method on the interface.

That being said, since this API isn't upstream and nothing depends on it, I'm okay with this change. We should just change the name to be `GetSignedUrl` though.